### PR TITLE
Added checks for power-states

### PR DESCRIPTION
**Pull request by m-hilgendorf**

### CODE REVIEW

**Review by bhusang — APPROVED** [review]
Looks good to me, some gnarly logic in power-state propagation which looks cool.

File: utils/property-structs.stanza [hunk 1]
```diff
@@ -128,3 +128,77 @@ public pcb-struct ocdb/property-structs/CrystalResonator :
...
+      val instance = containing-instance(pin) as Instance
+      if has-property?(instance.power-supply-component) and 
+         property(instance.power-supply-component) : 
+        do(visit{_, power-states}, pins(instance))
+
+  ; propagate the power-states to connected power-pins
+  do(visit{_0, property(_0.power-states)}, roots)
+
+  ; propagate from power pins to io-pins
+  for io-pin in filter(is-digital-io?, all-pins) do :
```
Comment (bhusang) [inline utils/property-structs.stanza:191]: You need to add ocdb/checks to the package import list? Per CI, I think.

Comment (m-hilgendorf) [reply utils/property-structs.stanza]: Fixed! I moved the function definition to `ocdb/checks`, which makes more sense to me and doesn't expose a private API

**Review by bhusang — APPROVED** [review]
Looks good!

**Pull request merged**